[Hue Koh] iP

huekoh opened the pull request:
# GMO (GameMate Organizer) 👾 🚀
> "Who wants to play video games?" - GMO

**GMO's not a bad influence... he**
- helps you remember your tasks
- ~SULKS~ _reminds_ you when you are slaved by work
- tells you to hurry so you can play with him!

**All you need to do is**
1. Download GMO [here](https://github.com/huekoh/ip)
2. Double click it
3. Start telling GMO your dues
4. Let him keep track of them for you 🤩

And he is **FREE**

GMO's Features:
- [X] Keep track of tasks (todos, dues, events)
- [X] Manage task status
- [X] Sort tasks by date
- [ ] Tell you to skip lecture

And if you are a Java programmer, use him to practice your coding too! Here's GMO's `constructor` method:
``` ruby
public GMO() {
        ui = new Ui();
        storage = new Storage();

        try {
            tasks = new TaskList(storage.loadData(), ui, storage);
        } catch (IOException e) {
            System.out.println("Error: Unable to load data. " + e.getMessage());
            tasks = new TaskList(ui);
        }
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

huekoh committed:
Created BMO.java, my own chatbot

huekoh committed:
Updated BMO.java

BMO now echoes back everything you say to him

huekoh committed:
Implemented list adder

huekoh committed:
implemented checklog array

huekoh committed:
created Task.java

abstracted tasks into a separate class

huekoh committed:
Created more .java files

huekoh committed:
created subclasses of task

huekoh committed:
Implemented text ui testing

huekoh committed:
Improved error handling

Error handling now handles these:
1. general error
2. index provided is out of range
3. task to be done is already done
4. task to be redone is already not done
5. no index was provided
6. user input add with nothing following up
7. incorrect deadline input format
8. incorrect todo input format
9. incorrect event input format

huekoh committed:
added task deletion option

huekoh committed:
Added save and load of tasks data

huekoh committed:
Update BMO.java addLog function

The formatter now recognises date time strings with "/" to express dates as valid. Previously, strings containing "/" would throw a format error and not accept it as a valid string

huekoh committed:
Create BMO.java formatDateTime function

When user adds a task with a date time variable, the string will be formatted into a LocalDateTime object that is stored in the task object.

huekoh committed:
Merge branch 'branch-level-8'

nichee commented on src/main/java/BMO.java:
```diff
@@ -0,0 +1,321 @@
+    static void viewLog() {
+        StringBuilder logPrint = new StringBuilder();
+        if (taskLog.isEmpty()) {
+            System.out.println(Constants.emptyLogPrint);
```
Consider renaming this function to "printEmptyLog" to start the name with the verb

nichee commented on src/main/java/BMO.java:
```diff
@@ -0,0 +1,321 @@
+        int index = Integer.parseInt((input));
+
+        if (index > taskLog.size() || index <= 0) {
+            System.out.println(Constants.errorPrint.outOfRange());
```
Consider changing the naming convention of the printing functions like outOfRange, perhaps use a verb?

nichee commented on src/main/java/ToDos.java:
```diff
@@ -0,0 +1,11 @@
+public class ToDos extends Task {
```
As far as I can tell this is the Todo class and not multiple Todos? Perhaps this can be changed to Todo since its used to instantiate single instances, so "ToDos" so it may be confusing

nichee reviewed (commented):
All good! Just small naming issues here and there

qinboan commented on src/main/java/BMO.java:
```diff
@@ -0,0 +1,321 @@
+import java.io.IOException;
+import java.nio.file.Files;
+import java.time.LocalDateTime;
+import java.util.*;
+import java.lang.*;
```
Avoid using wildcard imports as per java coding standard!

qinboan commented on src/main/java/BMO.java:
```diff
@@ -0,0 +1,321 @@
+            String taskDescription = info[2].trim();
+
+            switch (taskType) {
+                case "T":
```
There should be no indentation for case clauses

qinboan commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,31 @@
+        return (isDone ? "[X]" : "[ ]"); // mark done task with X
+    }
+
+    public Boolean getStatus() {
```
Consider renaming this boolean method to sound like booleans.
```suggestion
    public Boolean isDone() {
```

qinboan commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,90 @@
+        return taskLog;
+    }
+
+    // Other methods for task management...
```
Maybe you should delete this comment as it serves no purpose :)

qinboan reviewed (commented):
LGTM, just a few nits to fix!

huekoh committed:
Abstract data manager out

huekoh committed:
Abstract data manager out

HusseinSafwan02 commented on src/main/java/Constants.java:
```diff
@@ -0,0 +1,101 @@
+            + "    Let's play mario kart right now!!\n"
+            + "----------------------------------------------------------------------------------\n";
+
+    public static class errorPrint {
```
maybe you can consider renaming the class to errorPrints, since it has multiple error messages

HusseinSafwan02 commented on src/main/java/TaskManager.java:
```diff
@@ -0,0 +1,90 @@
+    }
+
+    private boolean isValidContentFormat(String content) {
+        // Implement your logic to check whether the content format is valid
```
I'm not very sure of the purpose of the comment here

HusseinSafwan02 commented on src/main/java/BMO.java:
```diff
@@ -0,0 +1,321 @@
+
+    static void parseData(String content) {
+        String[] lines = content.split("\n");
+        Integer indexCounter = 0;
```
you can use int here else there is unnecessary unboxing

HusseinSafwan02 reviewed (commented):
Just some minor fixes, well done!

huekoh committed:
Abstracted out UI, Parser, TaskList, Commands, and Storage

huekoh committed:
Abstracted out UI, Parser, TaskList, Commands, and Storage

huekoh committed:
Troubleshooting

huekoh committed:
Troubleshooting

huekoh committed:
Ensured errors don't crash terminal

huekoh committed:
Ensured errors don't crash terminal

huekoh committed:
Merge pull request #1 from huekoh/A-MoreOOP

A more oop

huekoh committed:
Revert "Merge pull request #1 from huekoh/A-MoreOOP"

This reverts commit 16a7afd8fbf2f80949fa3a2f8c2d0914a3db1e79, reversing
changes made to b3a931f3a430875d9b4406d9184252976c5ea485.

huekoh committed:
Removed redundant imports

huekoh committed:
Removed redundant imports

huekoh committed:
Merge branch 'A-MoreOOP'

huekoh committed:
Revert "Merge branch 'A-MoreOOP'"

This reverts commit 5f73efc7a974e03fc4659b29360d5e5eb49bfc1a, reversing
changes made to e5370f582dfffb0c47d02be41db00168461e789e.

huekoh committed:
Organised classes into java packages

huekoh committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

huekoh committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

huekoh committed:
Merge branch 'A-Gradle'

huekoh committed:
Added JUnit testing for Storage.java and Parser.java

huekoh committed:
Merge branch 'JUnit'

huekoh committed:
Updated relative path of data file in Storage.java

huekoh committed:
Implemented JAR

huekoh committed:
Added Java documentation

huekoh committed:
Added more java documentation

huekoh committed:
Fomatted java files according to Google Coding Style

huekoh committed:
Update DueCommand.java

huekoh committed:
Merge branch 'A-CodingStandard'

huekoh committed:
Fomat to google java code style

huekoh committed:
Implemented find task by keyword

huekoh committed:
Merge branch 'Level-9'

huekoh committed:
Implemented Checkstyle

huekoh committed:
Fixed code to adhere to checkstyle test

huekoh committed:
Updated GUI up to JavaFX tutorial 2

huekoh committed:
Updated GUI to have user interaction

huekoh committed:
Updated GUI code to use FXML, still buggy

huekoh committed:
Updated all UI.class methods to return a String

huekoh committed:
Updated execute() of command to return a string for response

huekoh committed:
Removed separation lines from UI

huekoh committed:
Removed unused methods from BMO.java

huekoh committed:
Add assertions

Add assertions to the BMO, Storage and TaskList classes

Several methods in these classes require important assumptions to hold true in order for the program to run.

Assertions for these conditions to the code helps the debugging process.

Let's add assertions to these classes.

huekoh committed:
Update Parser.java

Parse parse method is too long and handles both input type checking and input format checking.

Having smaller methods to handle the different types of possible inputs makes the code more readable.

Let's update the parser method to handle only input type checking, and abstract out input format checking to appropriate methods.

huekoh committed:
Update Parser.java

Parse parse method is too long and handles both input type checking and input format checking.

Having smaller methods to handle the different types of possible inputs makes the code more readable.

Let's update the parser method to handle only input type checking, and abstract out input format checking to appropriate methods.

huekoh committed:
Add assertions

Add assertions to Deadlines and Events classes

huekoh committed:
Refactor TaskList.java constructor

TaskList constructor method is too long and has duplicated code.

Swapping around the try block to encapsulate the switch block instead of vice versa helps remove duplicate code and improve readability.

Let's update the TaskList constructor method and reduce duplicate codes

huekoh committed:
Merge pull request #3 from huekoh/A-Assertions

A-assertions

huekoh committed:
Merge branch 'master' into A-CodeQuality

huekoh committed:
Merge pull request #4 from huekoh/A-CodeQuality

A-code quality

huekoh committed:
Add Ui.png

huekoh committed:
Add compareTo methods for task classes

huekoh committed:
Add sort command to sort task log

huekoh committed:
Update Ui.java with new method printSortEmptyLog()

huekoh committed:
Update README.md

huekoh committed:
Rebrand BMO to GMO

huekoh committed:
Credit code reuses

huekoh committed:
Update Gmo.java javadocs

huekoh committed:
Update README.md

huekoh committed:
Fix checkstyle violations

huekoh committed:
Update assets

huekoh committed:
Update README.md with Ui.png
